invenio_app_rdm/theme/assets/js/react_invenio_deposit/DepositController.js

```diff
@@ -20,37 +24,48 @@ export class DepositController {
     console.log("Validate record", record);
   }
 
-  async save_draft(record) {
+  async save_draft(record, { formik, store }) {
     // Saves a draft of the record
-    let payload = record;
+    const recordSerializer = store.config.recordSerializer;
+    let payload = recordSerializer.deserialize(record);
     this.validate(record);
-    if (!this.exists(record)) {
-      payload = await this.apiClient.create(record);
-      if (payload.errors) {
-        console.log("create errors", payload.errors);
-      } else {
+    try {
+      if (!this.exists(record)) {
+        payload = await this.apiClient.create(payload);
         const newURL = payload.data.links.edit;
         window.history.replaceState(undefined, "", newURL);
       }
+      const response = await this.apiClient.save(payload.data);
+      store.dispatch({
+        type: SAVE_SUCCESS,
+        payload: response,
+      });
+      formik.setSubmitting(false);
+    } catch (error) {
+      store.dispatch(setFormErrorsFromResponse(error, formik));
```

Comment:
Normal:

1- We should be able to save a network call (when save is a network call) by doing:
```js
const response = null;
if (!this.exists(record)) {
  response = await this.apiClient.create(payload);
  // ...
} else {
  response = await this.apiClient.save(payload.data);
}
```

because we already pass the record in `create` .

2- the try-catch allows us to catch network errors, but not invalid metadata errors, right? We still need to parse the response to determine if there were errors and dispatch SAVE_SUCCESS accordingly.

3- Since we only deal with publish flow (in this issue and for the may release), we could just disable all the save section and focus on the publish section below only.

Reply:
@fenekku If the call resolves to a ValidationError then we catch it but if the response is 200, then yes we need to parse it. But shouldn't we avoid this?

Reply:
@fenekku 
1- I don't see how we can save a network call as if the record doesn't exist we need to create it first before we save it. Currently, because we are using the `/records` API the save doesn't exist so we mock it but this will not happen when we integrate the `/drafts` API.
2- The `try/catch` is working properly if the network requests leads to an error response code. `ValidationError` gives a 400 so we are able to catch it.
3- I will disable in the UI the `save` button!

Reply:
1- I interpreted https://codimd.web.cern.ch/w5jnQsJ3Sn6Ne5LLEQD0hg#Create-draft-initial-Deposit-page-Auto-save as: the API accepts creation of a draft by passing form values as initial values directly (no need to create something empty first). This applies because we are in the context of the draft (Save). In the context of publishing, we indeed need to make sure a record exists before. We would create it (with the values from the form as initial values) and then publish (no need to "save" it after it was created though). 

2- wow axios does this for us?! That's nice.

3- :heavy_check_mark:

Reply:
@fenekku sorry I missed that context :) Yes in that case we can change a bit the workflow so we can save one request :)